packages/origin-ui-core/src/components/Organization/Organization.tsx

```diff
@@ -20,24 +22,48 @@ export const roleNames = {
 };
 
 export function Organization() {
-    const userOffchain = useSelector(getUserOffchain);
+    const user = useSelector(getUserOffchain);
+    const dispatch = useDispatch();
+    const offChainDataSource = useSelector(getOffChainDataSource);
+    const storedOrganizations = useSelector(getOrganizations);
+    const [fetching, setFetching] = useState<boolean>(false);
+
+    const fetchOrganizations = async () => {
+        setFetching(true);
+        try {
+            const organizations: IOrganizationWithRelationsIds[] = await offChainDataSource.organizationClient.getAll();
+            dispatch(addOrganizations(organizations));
+        } catch (error) {
+            console.error('Error fetching list of all organizations', error);
+        }
+        setFetching(false);
+    };
+
+    if (
+        user &&
+        (!storedOrganizations || storedOrganizations.length === 0) &&
+        !fetching &&
+        isRole(user, Role.Admin, Role.SupportAgent)
+    ) {
+        fetchOrganizations();
```

Comment:
Does this need to be fetched in this Organization component? I think this is only needed in the `OrganizationTable` component.

Reply:
You are right. Even in `OrganizationTable` its not needed as it accessible based on role